src/client/datascience/jupyter/kernels/kernelConnectionWrapper.ts

```diff
@@ -223,7 +271,7 @@ export class KernelConnectionWrapper implements Kernel.IKernelConnection {
         this.startHandleKernelMessages(this.kernel.session?.kernel);
     }
     private startHandleKernelMessages(kernelConnection: Kernel.IKernelConnection) {
-        this.kernelConnection = kernelConnection;
+        this._kernelConnection = kernelConnection;
```

Comment:
set _previousKernelConnection first? in case it changes more than once

Reply:
Updated the code to ensure `_previousKernelConnection` never undefined and readonly (as its initialized in the ctor)

Reply:
but it is _only_ set in the constructor, so if it changes more than once, it will no longer be the previous kernel connection. Should the variable be renamed to _initialKernelConnection?